### PR TITLE
add unity es6 blog post

### PR DESCRIPTION
**Pull request by depoulo**
- [x] @choffmeister and/or @Raigen please review from tech perspective.
- [x] marketing feedback (@steffimeier and @maletzke please)
- [x] native proof read (@JessTinker please)
- [x] probably amend the post date (@depoulo)

### CODE REVIEW

File: _posts/2015-11-11-full-stack-es6-with-webpack-and-babel.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import React from 'react';
+
+export default class ImagePlugin extends React.Component {
+  constructor() {
+    super();
+{% endhighlight %}
+
+Much leaner than good ol' RequireJS syntax, isn't it?
+The alert reader might notice some more ES6 novelties here: Yup, ES6 introduces the `class`, `extends`, and `super` keywords as syntactic sugar over JavaScript's prototypal inheritance system, something you might already have been using in other ECMAScript dialects like TypeScript (or, in my case, ActionScript 2).
+When our Java colleagues saw this the other day, they went like: **"Oh, this looks almost like Java!"**. Dear Jens, to a JS developer this is _not a compliment_ ;)
```
Comment (depoulo) [inline _posts/2015-11-11-full-stack-es6-with-webpack-and-babel.md]: @jensfischerhh I hope you don't mind

File: _posts/2015-11-11-full-stack-es6-with-webpack-and-babel.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  constructor() {
+    super();
+{% endhighlight %}
+
+Much leaner than good ol' RequireJS syntax, isn't it?
+The alert reader might notice some more ES6 novelties here: Yup, ES6 introduces the `class`, `extends`, and `super` keywords as syntactic sugar over JavaScript's prototypal inheritance system, something you might already have been using in other ECMAScript dialects like TypeScript (or, in my case, ActionScript 2).
+When our Java colleagues saw this the other day, they went like: **"Oh, this looks almost like Java!"**. Dear Jens, to a JS developer this is _not a compliment_ ;) 
+
+## But wait there's more
+Another much-awaited ES6 feature we have heavily adopted are **_arrow functions_ or _lambdas_**. One nice thing about them, especially for the Java developer new to JavaScript (hello Jens), is that unlike traditional JS functions they don't introduce their own `this`.
```
Comment (depoulo) [inline _posts/2015-11-11-full-stack-es6-with-webpack-and-babel.md]: @jensfischerhh I hope you don't mind

**Comment by depoulo** [comment]
_mentioning @steffimeier, @maletzke  and @JessTinker here again since I'm unsure if GitHub sends mail when a mention is added in a comment edit._

**Comment by depoulo** [comment]
comment from Birgit: "Ich kümmere mich um den Veröffentlichungstermin, wenn ich zurück bin. **Bitte vorher nicht mergen!**"

File: _posts/2015-11-11-full-stack-es6-with-webpack-and-babel.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
+---
+layout: post
+title: "On the way to full stack ECMAScript 6"
+date: "2015-11-16 12:00:00"
+icon: code
+categories: tech-stories
+authors: ["Paolo"]
+---
+
+**JavaScript is moving fast these days.** With all the languages out there compiling to it, JS itself is making a big leap forwards with the standardisation of its next version called _ECMAScript 6_ (or _ES6_, or _ECMAScript 2015_, or _ES2015_. [Naming is probably the hardest programming related task](https://twitter.com/a_ramella/status/656522461480099840), and if you think that _ECMAScript sounds like a skin disease_, you're [in good company with the inventor of JavaScript himself](https://mail.mozilla.org/pipermail/es-discuss/2006-October/000133.html)).
```
Comment (JessTinker) [inline _posts/2015-11-11-full-stack-es6-with-webpack-and-babel.md]: SHould we put a (JS) after JavaScript so people know that further in the doc when you write JS they know what you are talking about?

File: _posts/2015-11-11-full-stack-es6-with-webpack-and-babel.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+In this post, I want to share things we've learned during our ongoing adoption of the new language features.
+
+## JS @epages
+My team and I are working on an upcoming part of the ePages software that provides an easy to use drag-and-drop editor for the layout and appearance of an ePages shop, and serves the actual shop web site by displaying all product and editorial data in that layout.
```
Comment (JessTinker) [inline _posts/2015-11-11-full-stack-es6-with-webpack-and-babel.md]: I think the sentence is perhaps too long what do you think about breaking it up like this:

My team and I are working on an upcoming part of the ePages software that provides an easy to use drag-and-drop editor for the layout and appearance of an ePages shop. It serves the actual shop website by displaying all product and editorial data in that layout.

File: _posts/2015-11-11-full-stack-es6-with-webpack-and-babel.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+Under the hood this means aggregating and manipulating JSON data from various REST resources, and providing a live preview of the result in the layout edit mode. Both are tasks JavaScript is really good at, so my team does **JS all the way** from the (Mongo) database through the (Node) server down to the browser.
+
+## JavaScript modules revisited
+One of our pains as JavaScript developers has always been **the lack of a module system**. Node.js introduced the _CommonJS_ approach consisting of `module.exports` and `require('module')`, which works well on the server but not on the client. That's because in the browser, we have network latency, and thus loading a module is an asynchronous operation. When we started our project, _RequireJS_ solved this for us as elegant as it can get without introducing a compile step, using its _AMD (asynchronous module definition)_ syntax consisting of `define(['module'], function (module) {...});`.
```
Comment (JessTinker) [inline _posts/2015-11-11-full-stack-es6-with-webpack-and-babel.md]: using its _AMD (asynchronous module definition)_ syntax, consisting of `define(['module'], function (module) {...});`.

File: _posts/2015-11-11-full-stack-es6-with-webpack-and-babel.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+But we're writing _isomorphic_ JavaScript, which is fancy speaking for running (parts of) the exact same code on client and server alike. If you're doing this in combination with RequireJS, you can easily get a headache from looking at statements like `var require = require('requirejs');`.
```
Comment (JessTinker) [inline _posts/2015-11-11-full-stack-es6-with-webpack-and-babel.md]: But we're writing _isomorphic_ JavaScript, which is a fancy way of saying: running (parts of) the exact same code on client and server alike. 

(What do you think about that change?)

File: _posts/2015-11-11-full-stack-es6-with-webpack-and-babel.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+import ImmutablePropTypes from 'react-immutable-proptypes';
+import React from 'react';
+
+export default class ImagePlugin extends React.Component {
+  constructor() {
+    super();
+{% endhighlight %}
+
+Much leaner than good ol' RequireJS syntax, isn't it?
+The alert reader might notice some more ES6 novelties here: Yup, ES6 introduces the `class`, `extends`, and `super` keywords as syntactic sugar over JavaScript's prototypal inheritance system, something you might already have been using in other ECMAScript dialects like TypeScript (or, in my case, ActionScript 2).
```
Comment (JessTinker) [inline _posts/2015-11-11-full-stack-es6-with-webpack-and-babel.md]: something you might have already been using in other ECMAScript dialects like TypeScript (or, in my case, ActionScript 2).

File: _posts/2015-11-11-full-stack-es6-with-webpack-and-babel.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+When our Java colleagues saw this the other day, they went like: **"Oh, this looks almost like Java!"**. Dear Jens, to a JS developer this is _not a compliment_ ;)
+
+## But wait there's more
+Another much-awaited ES6 feature we have heavily adopted are **_arrow functions_ or _lambdas_**. One nice thing about them, especially for the Java developer new to JavaScript (hello Jens), is that unlike traditional JS functions they don't introduce their own `this`.
```
Comment (JessTinker) [inline _posts/2015-11-11-full-stack-es6-with-webpack-and-babel.md]: Another much-awaited ES6 feature we have heavily adopted is

File: _posts/2015-11-11-full-stack-es6-with-webpack-and-babel.md [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+    : {}
+};
+{% endhighlight %}
+
+Prior to ES6, the above code would have contained three times `function (...) {...}` and `return`, alongside some extra line breaks and curly braces. **CoffeeScript addicts will feel right at home!**
+Once again, the alert reader might notice that we're making use of two other awesome ES6 features in the above code: _Promises_ for dealing with asynchronous execution, and `Object.assign` for library-free data structure augmentation.
+
+But the list doesn't end here: [template strings](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/template_strings), `let` and `const`, and [Object literal extensions](https://github.com/lukehoban/es6features#enhanced-object-literals) are all things we're using in our client side code thanks to Babel and Webpack, and on the server thanks to V8/Node.js supporting them natively. That's right: **We're not yet using Babel on the server**, but once we've found the time to do that as well we'll address it in a follow-up post.
+
+## So where's the bad news?
```
Comment (JessTinker) [inline _posts/2015-11-11-full-stack-es6-with-webpack-and-babel.md]: So what's the bad news?

File: _posts/2015-11-11-full-stack-es6-with-webpack-and-babel.md [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+
+First of all, you'll need to update your tool set to support ES6. In our case, this meant switching to [an ES6 supporting fork](https://github.com/deedubs/es6-plato) of the _Plato_ complexity analysis tool, and turning on ES6 support in the _ESlint_ linter.
+Since code coverage measures shouldn't be affected by transpiler output, we had to include the [_Isparta_](https://github.com/douglasduteil/isparta) library, which, again via a fork, enables the _karma-coverage_ task to provide _Istanbul_ code coverage for ES6.
+
+Another issue is **debugging ES6 code**. The browsers' developer tools do a great job when it comes to displaying the original ES6 code in the source code view,  thanks to _source maps_. But they stop there. Scope variables currently don't use source mapping, which is why you'll still have to deal with transpiler output like `_react2.default` while inspecting code from within a breakpoint.
+
+Speaking of breakpoints: With all the beauty of one-liner arrow functions, I sometimes find myself adding line breaks and curly braces just to be able to set a breakpoint inside a `filter` predicate function.
```
Comment (JessTinker) [inline _posts/2015-11-11-full-stack-es6-with-webpack-and-babel.md]: With the beauty of one-liner arrow functions,

File: _posts/2015-11-11-full-stack-es6-with-webpack-and-babel.md (hunk 3 above)
Comment (depoulo) [inline _posts/2015-11-11-full-stack-es6-with-webpack-and-babel.md]: To be honest, people who don't get scared away by the headline will definitely know what JS stands for in this context.

I adopted all the other changes, it's always a pleasure to learn from a native speaker :)

Comment (choffmeister) [inline _posts/2015-11-11-full-stack-es6-with-webpack-and-babel.md]: True story :)

**Comment by depoulo** [comment]
@BirgitBader go for it!

**Pull request merged**